src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameRequestStream.cs

```diff
@@ -119,8 +119,19 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
             var task = ValidateState(cancellationToken);
             if (task == null)
             {
-                // Needs .AsTask to match Stream's Async method return types
-                return _body.ReadAsync(new ArraySegment<byte>(buffer, offset, count), cancellationToken).AsTask();
+                return _body.ReadAsync(new ArraySegment<byte>(buffer, offset, count), cancellationToken);
+            }
+            return task;
+        }
+
+        public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
+        {
+            // REVIEW: Is bufferSize a minimum, maximum or suggestion? Should we validate that it's <= 4096 or >= 4096?
+
```

Comment:
The default is 81920, so I don't think we can realistically reject calls with a bufferSize > 4096.